### ISSUE TITLE
Bug: Wildcard example is not working

### ISSUE THREAD

**Issue by daniesy** [express-validator/express-validator#943]
## Describe the bug
Wildcard rules don't seem to be checked.

## To Reproduce
Run the following code and make a request with an empty post:

    const express = require('express')
    const { check, sanitize, validationResult } = require('express-validator');

    const app = express()
    app.use(express.json());

    const port = 3001

    app.post('/', [
      check('addresses.*').isPostalCode(),
    ], (req, res) => {
      const errors = validationResult(req);
      if (!errors.isEmpty()) {
        return res.status(400).json({ errors: errors.array() });
      }

      res.send('Hello World!1')
    });

    app.listen(port, () => {
      console.log(`Example app listening at http://localhost:${port}`)
    })

## Expected behavior
I expected some validation errors to pop out.

## Current behavior
No errors.

## Express-validator version:
 - Version: 6.6.1

**Comment by fedeci** [comment]
Hi, can you add also the body of the failing request, please?
It would definitely help to understand better where the error comes from.

**Comment by meircarlos** [comment]
I didn't try this particular case, but I have been trying to do the same with checkSchema and it doesn't work either.

I tried something like the following, but it doesn't give any errors:
```
checkSchema({
    'names.*': {
        in: 'body',
        exists: true
    }
}
```

So instead what I did is:
```
checkSchema({
    names: {
        in: 'body',
        isLength: {
            options: {
                 min: 1
            },
            errorMessage: 'There should be at least 1 name'
        },
    }
}
```

**Comment by fedeci** [comment]
Without the request body I still understand nothing about the error :)

**Comment by meircarlos** [comment]
The request body can be empty, the bug is that it's not complaining about `req.body.names` not being provided

**Comment by fedeci** [comment]
`isLength` only validates strings or the first element of the array provided.
To validate the array length you must use `isArray`.

e.g.
```js
body('names').isArray({ min:1 })
```

**Comment by meircarlos** [comment]
Great, thanks for the update.

A quick side question though, is there a way to validate that the length of 2 different field arrays are the same using `checkSchema`?

**Comment by fedeci** [comment]
Sure, but you have to use a custom validator :)
